File: simple_automations/os_operations/python/batch_renamer.py
import os

# to be replaced string and file extension filter
search = "file"
replace = "document"
type_filter = ".py"

# get all files from current directory
dir_content = os.listdir('.')
docs = [doc for doc in dir_content if os.path.isfile(doc)]
renamed = 0

print(f"{len(docs)} of {len(dir_content)} elements are files.")

# go through all the files and check if they match the search pattern
for doc in docs:
    # separate name from file extension
    doc_name, filetype = os.path.splitext(doc)

    # filter for files with the right extension
    if filetype == type_filter:
        # check if search text is in doc name
        if search in doc_name:
            
            #replace with the given text
            new_name = doc_name.replace(search, replace) + filetype
            os.rename(doc, new_name)
            renamed += 1

            print(f"Renamed file {doc} to {new_name}")

print(f"Renamed {renamed} of {len(docs)} files.")

File: simple_automations/os_operations/javascript/batch_renamer.js
const fs = require('fs');
const path = require('path');

const type_filter = '.txt';
const search = 'file';
const replace = 'document';

// filter the current directory for files
const dir = fs.readdirSync('.');
const files = dir.filter(f => fs.statSync(f).isFile());

console.log(`${files.length} of ${dir.length} elements are files.`);

let renamed = 0;

for (const doc of files) {
  const file = path.parse(doc);

  // skip not matching file types
  if (file.ext != type_filter) continue;
  // skip not matching file names
  if (!file.name.includes(search)) continue;

  // rename the actual file
  fs.renameSync(doc, doc.replace(search, replace));
  renamed++;
}

console.log(`renamed ${renamed} of ${files.length} files.`);